Bottom: 8d6bf8641096a3c39d78d518352add5d102babf1
Top:    a0c0710594de11aa2175f35ae37d624c5ac63efb
Author: Lawrence D'Anna <lawrence_danna@apple.com>
Date:   2019-10-05 15:08:15 -0700

fix-convert-sbdebugger-filehandle


---
diff --git a/lldb/packages/Python/lldbsuite/test/python_api/file_handle/TestFileHandle.py b/lldb/packages/Python/lldbsuite/test/python_api/file_handle/TestFileHandle.py
index 6c6d2eef6b6..a20b0c63ff6 100644
--- a/lldb/packages/Python/lldbsuite/test/python_api/file_handle/TestFileHandle.py
+++ b/lldb/packages/Python/lldbsuite/test/python_api/file_handle/TestFileHandle.py
@@ -841,12 +841,19 @@ class FileHandleTestCase(lldbtest.TestBase):
             self.debugger.SetOutputFileHandle(None, False)
             self.debugger.SetErrorFileHandle(None, False)
             sbf = self.debugger.GetOutputFile()
-            self.assertEqual(sbf.GetFile().fileno(), 1)
+            if sys.version_info.major >= 3:
+                # python 2 lacks PyFile_FromFd, so GetFile() will
+                # have to duplicate the file descriptor and make a FILE*
+                # in order to convert a NativeFile it back to a python
+                # file.
+                self.assertEqual(sbf.GetFile().fileno(), 1)
             sbf = self.debugger.GetErrorFile()
-            self.assertEqual(sbf.GetFile().fileno(), 2)
+            if sys.version_info.major >= 3:
+                self.assertEqual(sbf.GetFile().fileno(), 2)
         with open(self.out_filename, 'r') as f:
             status = self.debugger.SetInputFile(f)
             self.assertTrue(status.Success())
             self.debugger.SetInputFileHandle(None, False)
             sbf = self.debugger.GetInputFile()
-            self.assertEqual(sbf.GetFile().fileno(), 0)
+            if sys.version_info.major >= 3:
+                self.assertEqual(sbf.GetFile().fileno(), 0)
